### PR TITLE
Tied Scores Support from ScoreManager

### PR DESCRIPTION
**Pull request by DJSteins**
Updates the ScoreManager with a method that gets the leader board positions. Takes into account players with the same score to return tied positions.

### CODE REVIEW

File: Assets/_PartyParkade/Tests/ScoreManager_Tests.cs [hunk 1]
```diff
@@ -0,0 +1,97 @@
+using System.Collections;
```
Comment (m4rtint) [inline Assets/_PartyParkade/Tests/ScoreManager_Tests.cs]: Can you also test 
scores[Player.Player_1] = 0;
scores[Player.Player_2] = 1;
Assert Placement

Change the score again
scores[Player.Player_1] = 1;
scores[Player.Player_2] = 1;

And then Assert Placement again

Comment (DJSteins) [reply Assets/_PartyParkade/Tests/ScoreManager_Tests.cs]: image

Comment (DJSteins) [reply Assets/_PartyParkade/Tests/ScoreManager_Tests.cs]: Oops just fixed the typo in that last method.

**Comment by mikeslgoh** [comment]
[Update data structure to take into account tied scores](https://trello.com/c/4aWj7LJK/409-update-data-structure-to-take-into-account-tied-scores)

**Comment by mikeslgoh** [comment]
[Refactor ScoreManager for testing](https://trello.com/c/m6Rz8MyC/315-refactor-scoremanager-for-testing)

**Pull request merged**